parquet/src/test/java/org/apache/iceberg/parquet/TestVariantReaders.java

```diff
@@ -1086,14 +1387,22 @@ private static GroupType field(String name, Type shreddedType) {
         .named(name);
   }
 
+  private static GroupType list(Type shreddedType) {
+    return Types.optionalList()
```

Comment:
When the element type is passed in, I think it will be more straightforward to construct this:

```java
Types.optionalList()
        .element(elementType)
        .named("typed_value");
```